### ISSUE TITLE
implement sharing to user-defined groups

### ISSUE THREAD

**Issue by atroxix** [owncloud/contacts#1100]
Users often work together in "working groups" and thus need to share files between each other. As users in groups may change it does make sense to define such group of users under Contacts app and use this group to share files. Currently files can be shared to groups defined by administrators in Users interface, but not to groups defined in Contacts app.

**Comment by babelouest** [comment]
This feature request is interesting but you mix different aspects of owncloud:

Contact app is not owncloud user list. If your contact list is the same list as the owncloud user list, it is defined by you and for your usage, even if those 2 lists can be related, for example with the ldap backend, they are completely different lists according to owncloud core system.

If you wanted to relate your contact lists groups with the share groups property of owncloud different apps, you have to create an engine that will try to match every owncloud users with every contact in your addressbooks and extract the group name. I find it very much overkill...

Anyway if you really want/need this feature, I suggest you make this feature request in owncloud core development git, not the contact app.

**Comment by atroxix** [comment]
Indeed, it seems that the only way to share anything to users in Contacts app is by using Federated ID. So in my case what we want is to:
1. Define groups that have both local and remote (ie. Federated ID) OwnCloud users in them
2. Share to those groups

It seems to me that if this were somehow to be implemented in OwnCloud Core then it would require a way to define users with Federated ID in Core too, ie. the same functionality that is implemented in Contacts app. I suspect that this won't be implemented in Core, and if this is true then it would never be possible to define such groups by Core functionality.

Based on the thoughts above it feels that Contacts app would still be the best place to implement it. One simple way to do it would be to make it possible to include OwnCloud local users in contact groups. Then the sharing functionality would need to be extended from matching users with defined Federated IDs to matching also groups.

Does that make sense? :)

**Comment by babelouest** [comment]
It makes some sense, although I don't know the Federated ID field so I'm not an expert.
Anyway, as you say:

> I suspect that this won't be implemented in Core

You'll never know for sure unless you ask for it. You can also start an implementation if you're willing.

**Comment by atroxix** [comment]
> It makes some sense, although I don't know the Federated ID field so I'm not an expert.

This is interesting, considering the fact that it's possible to define Federated ID for a contact in Contacts app, and this contact can then be specified as a receiver of a share.

But if this is really a coincidence of some functionality in OwnCloud Core and not implementation in Contacts app then I suggest you take a look at https://owncloud.org/federation/

**Comment by atroxix** [comment]
This is at least partially already requested in https://github.com/owncloud/core/issues/8201
